app/uk/gov/hmrc/pillar2/repositories/RegistrationCacheRepository.scala

```diff
@@ -81,58 +65,53 @@ class RegistrationCacheRepository @Inject() (
       collectionName = "user-answers-records",
       mongoComponent = mongoComponent,
       domainFormat = RegistrationDataEntryFormats.format,
-      extraCodecs = Seq(
-        Codecs.playFormatCodec(JsonDataEntry.format),
-        Codecs.playFormatCodec(DataEntry.format)
-      ),
       indexes = Seq(
         IndexModel(
-          Indexes.ascending(expireAtKey),
+          Indexes.ascending(lastUpdatedKey),
           IndexOptions()
-            .name("dataExpiry")
-            .expireAfter(2419200, TimeUnit.SECONDS)
-            .background(true)
+            .name("lastUpdatedIndex")
+            .expireAfter(config.defaultDataExpireInDays, TimeUnit.DAYS)
+        ),
+        IndexModel(
+          Indexes.ascending("id"),
+          IndexOptions().name("id").unique(true).background(false)
         )
-      ),
-      replaceIndexes = true
+      )
```

Comment:
replace Indexes is required otherwise it may fail in environment.

Reply:
added back